src/swap/utils/processSwapTransaction.ts

```diff
@@ -92,17 +87,24 @@ export async function processSwapTransaction({
     value: transaction.value,
     data: transaction.data,
   });
-  await Promise.resolve(onStart?.(txHash));
-
+  setLifecycleStatus({
+    statusName: 'transactionPermitApproved',
+    statusData: {
+      transactionHash: txHash, // Check with Alec what this should be
+    },
+  });
```

Comment:
we can probably omit this - the only difference between this state and `statusName: 'success'` is the former is when the swap transaction is submitted and the latter is when the swap transaction is confirmed onchain, it should be less than 1 second between both. if developers ask for this we can add a `statusName: 'swapSubmitted'` or similar state

```suggestion

```